core/src/test/scala/pureconfig/PureconfSuite.scala

```diff
@@ -619,24 +620,22 @@ class PureconfSuite extends FlatSpec with Matchers with OptionValues with TryVal
     case class SampleConf(a: Int, b: String)
     loadConfig[SampleConf](conf).failure.exception shouldEqual KeyNotFoundException("a")
 
-    implicit val mapping = new ConfigFieldMapping[SampleConf] {
-      def apply(fieldName: String) = fieldName.toUpperCase
-    }
+    implicit val productHint = ProductHint[SampleConf](ConfigFieldMapping(_.toUpperCase))
 
     loadConfig[SampleConf](conf) shouldBe Success(SampleConf(2, "two"))
   }
 
   it should "allow customizing the field mapping with word delimiters" in {
```

Comment:
Well, the intention of this test is not to test specifically the camel case conversion - that is done extensively in `NamingConventionTest.scala`. This one is for testing that adding an appropriate hint allows different field mappings to be applied. However, I can replace "word delimiters" with "different naming conventions" and test more than one mapping. What do you think?